docs/authentication-rdd.md

```diff
@@ -0,0 +1,406 @@
+
+#### 1.2 Magic Link Reuse Prevention
+**Requirement**: Given a user attempting to use the same magic link twice
+
+**Behavior**:
+- Display message: "This link has already been used"
+- Offer two options:
+  1. "Sign in with passkey" (if passkey exists)
+  2. "Send new magic link"
+
+### 2. Sign-In Flow
+
+#### 2.1 Primary Authentication (Passkey)
+**Trigger**: User visits sign-in page or attempts to access protected content
+
+**Flow**:
+1. **Passkey Prompt**
+   - Display "Sign in with passkey" as primary option
+   - Show browser's WebAuthn authentication dialog
+   - Verify passkey against stored credentials
+   
+2. **Successful Authentication**
+   - Redirect to course content landing page
+   - Set authentication session
+   - Update last sign-in timestamp
+
+**Fallback Option**:
+- Link: "Don't have your passkey? Sign in with email"
+
+#### 2.2 Fallback Authentication (Magic Link)
+**Trigger**: User clicks "Sign in with email" fallback option
+
+**Flow**:
+1. **Email Collection**
+   - Prompt for email address
+   - Validate email exists in system
+   
+2. **Magic Link Dispatch**
+   - Send magic link to email
+   - Link expires after 15 minutes
+   - Link is single-use only
+   
+3. **Magic Link Authentication**
+   - User clicks magic link
+   - Validate link authenticity and expiration
+   - Authenticate user session
+   
+4. **Post-Authentication Flow**
+   - If user has existing passkey: Redirect to course content
+   - If user has no passkey: Prompt for passkey setup (optional)
+   - Redirect to course content landing page
+
+**Error Handling**:
+- Email not found → Redirect to sign-up flow
+- Magic link expired → Offer to resend new link
+- Invalid link → Show error with option to request new link
+
+### 3. Device Management Flow
+
+#### 3.1 New Device Access
+**Scenario**: Existing user wants to access account from new device
+
+**Flow**:
+1. User attempts passkey authentication → No passkey available
+2. User clicks "Sign in with email"
+3. Complete magic link authentication
+4. After successful auth, prompt: "Set up passkey for this device?"
+5. Optional passkey registration for new device
+
+#### 3.2 Email Change Flow
+**Scenario**: User wants to change their email address
+
+**Requirements**:
+- Must authenticate with current passkey OR current email magic link
+- Send verification to new email address
+- Maintain existing passkeys after email change
+- Update email in user profile
+
+## Technical Requirements
+
+### 3.1 Security Requirements
+
+#### Passkey Implementation
+- **Standard**: WebAuthn Level 2 compliance
+- **Authenticator Requirements**: 
+  - Platform authenticators preferred (Touch ID, Face ID, Windows Hello)
+  - Cross-platform authenticators supported (security keys)
+- **User Verification**: Required for all passkey operations
+- **Attestation**: None required (for broader compatibility)
+
+#### Magic Link Security
+- **Expiration**: 15 minutes maximum
```

Comment:
Bro magic links can take more than 10-15 minutes to be delivered. This should be hours, not minutes.